### PR TITLE
RFC: Add documentation for givens, ldltfact!, and eigvals!. Fixes #10526

### LINKED ISSUE #10526: a couple undocumented linalg functions
These are exported from Base and not documented:

* `eigvals!`
* `ldltfact!`
* `givens`

### PR DESCRIPTION
**Pull request by andreasnoack**
I think this shouldn't be merged before the other doc changes are completed.

Please check for bad language, typos and errors.

### CODE REVIEW

File: base/linalg/givens.jl [hunk 1]
```diff
@@ -227,7 +233,12 @@ function givens{T}(f::T, g::T, i1::Integer, i2::Integer)
     c, s, r = givensAlgorithm(f, g)
     Givens(i1, i2, convert(T, c), convert(T, s)), r
 end
+"""Computes a Givens rotation
+
+`givens{T}(A::AbstractArray{T}, i1::Integer, i2::Integer, col::Integer) -> (Givens{T}, r)`
 
+such that `G*A[:,col]=y` with `y[i1]=r`, and `y[i2]=0`. The cosine and sine of the rotation angle can be extracted from the `Givens` type with `G.c` and `G.s` respectively. The arguments `f` and `g` can be either `Float32`, `Float64`, `Complex{Float32}`, or `Complex{Float64}`. The `Givens` type supports left multiplication `G*A` and conjugated transpose right multiplication `A*G'`. The type doesn't have a `size` and can therefore be multiplied with matrices of arbitrary size as long as `i2<=size(A,2)` for `G*A` or `i2<=Size(A,1)` for `A*G'`.
```
Comment (kshyatt) [inline base/linalg/givens.jl]: `i2<=Size(A,1)` should be `size`, right?

Comment (mschauer) [inline base/linalg/givens.jl]: Same above.

File: base/linalg/givens.jl [hunk 2]
```diff
@@ -220,6 +220,12 @@ function givensAlgorithm{T<:AbstractFloat}(f::Complex{T}, g::Complex{T})
     return cs, sn, r
 end
 
+"""Computes a Givens rotation
+
+`givens{T}(f::T, g::T, i1::Integer, i2::Integer) -> (Givens{T}, r)`
```
Comment (mschauer) [inline base/linalg/givens.jl]: maybe `-> (G::Givens{T}, r)`

**Comment by kshyatt** [comment]
Now that https://github.com/JuliaLang/julia/pull/12835 is merged, is this ready to go?

**Comment by yuyichao** [comment]
If these are public doc, they should follow the format of other doc in `helpdb.jl` and add the function signature to the relavant section in the rst doc

**Comment by andreasnoack** [comment]
@yuyichao Is this process described anywhere? I don't really understand the new system. Would you explain "..add the function signature to the relavant section in the rst doc"?

**Comment by KristofferC** [comment]
The documentation strings should look like this, where the method signature is intented as the line in the doc string:

```julia
doc"""
    chol(A, [LU]) -> F

Compute the Cholesky factorization of a symmetric positive definite matrix `A`
and return the matrix `F`. If `LU` is `Val{:U}` (Upper), `F` is of type `UpperTriangular`
and `A = F'*F`. If `LU` is `Val{:L}` (Lower), `F` is of type `LowerTriangular` and 
`A = F*F'`. `LU` defaults to `Val{:U}`.
"""
```

In the `.rst` file where the doc should go there should be a corresponding entry, like:

```
.. function:: chol(A, [LU]) -> F
```

The documentation will then automatically be inserted in the `.rst`-file when `genstdlib.jl` is run.

**Comment by andreasnoack** [comment]
@KristofferC Thanks. We better write that down somewhere at some point.

**Comment by yuyichao** [comment]
I thought @jakebolewski put it in `CONTRIBUTE.md` but I guess it doesn't include the part about adding new function docs. Since I don't see that the `genstdlib` hack is going away anytime soon, I agree it's probably better to add this too...

**Comment by JeffBezanson** [comment]
This is one of 2 remaining 0.4 issues. Is this release blocking? Please merge or remove the milestone as soon as practical.

**Comment by andreasnoack** [comment]
Okay. I think this one is ready for review now. To counter comments on the lack of hard newlines: I still think new lines should be soft because

1. it allows users to have the terminal width that suits them without getting an ugly combination hard and soft new lines
2. you can edit the text without adjust all lines in a paragraph.
3. markup features result in a different line lenght in source and rendered text anyway

If there is now a consensus for hard newlines and if the parsing of newlines works then I'll of course update the pr.

**Comment by KristofferC** [comment]
A new line in the doc string doesn't count as a hard newline does it? I thought markdown didn't care about single new lines. Does the in-terminal-help hard break the line on single new lines in doc strings?

**Comment by tkelman** [comment]
Github refuses to wrap long lines in julia code. Until that changes we should wrap long lines for review and readability.

**Comment by andreasnoack** [comment]
@KristofferC you are right. So it is only a matter of formatting the source.

@tkelman I don't wan't this pr to take much longer so I've hit enter a couple of times in the source. However, I've just realized that GitHub's split mode actually wraps lines so if you switch to that you'll probably see a nice altenation between soft and hard newlines where every second line has one word.

**Pull request merged**